cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -961,6 +992,17 @@ private V1PodSpec createPodSpec(Type resourceType, Location runtimeConfigLocatio
     volumeMounts.add(new V1VolumeMount().name("workdir").mountPath(workDir));
     volumeMounts.addAll(Arrays.asList(extraMounts));
 
+    // Add workload identity volume, volume mount, and environment variable if applicable.
+    if (workloadIdentityServiceAccountEmail != null && !workloadIdentityServiceAccountEmail.isEmpty()) {
+      String ksaTTL = cConf.get(KubeMasterEnvironment.WORKLOAD_IDENTITY_SERVICE_ACCOUNT_TOKEN_TTL_SECONDS);
```

Comment:
instead of passing the cConf all the way down the stack, we can do this lookup in the constructor once. cConf just comes from masterEnvContext

Reply:
Fixed